Description: Use dpkg-architecture in cups-config to make it architecture-independant
 Also filter -L/usr/lib/(triplet) out of LDFLAGS.
 Also strip unnecessary libs from --libs.
 Also launch krb5-config at runtime to avoid getting usr/lib/(triplet) from krb5-config
Bug-Debian: https://bugs.debian.org/741519
Bug-Debian: https://bugs.debian.org/727058
Bug-Debian: https://bugs.debian.org/730838
Bug-Debian: https://bugs.debian.org/751157
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2014-06-11
--- a/cups-config.in
+++ b/cups-config.in
@@ -22,8 +22,8 @@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
 includedir=@includedir@
-libdir=@libdir@
-imagelibdir=@libdir@
+libdir=${prefix}/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)
+imagelibdir=${prefix}/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)
 datarootdir=@datadir@
 datadir=@datadir@
 sysconfdir=@sysconfdir@
@@ -35,7 +35,8 @@
 # flags for C++ compiler:
 CFLAGS=""
 LDFLAGS="@EXPORT_LDFLAGS@"
-LIBS="@LIBGSSAPI@ @EXPORT_SSLLIBS@ @LIBZ@ @LIBS@"
+LIBGSSAPI=`[ ! -x /usr/bin/krb5-config ] || /usr/bin/krb5-config --libs`
+LIBS="$LIBGSSAPI @EXPORT_SSLLIBS@ @LIBZ@ @LIBS@"
 
 # Check for local invocation...
 selfdir=`dirname $0`
@@ -50,7 +51,7 @@
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64 -a $libdir != /usr/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH); then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
@@ -105,7 +106,7 @@
 	    ;;
 	--libs)
 	    if test $static = no; then
-	        libs="@EXTLINKCUPS@ $LIBS";
+	        libs="@EXTLINKCUPS@";
  	        if test $image = yes; then
  	            libs="@EXTLINKCUPSIMAGE@ $libs"
 		fi
